### ISSUE TITLE
mvn clean install doesn't work

### ISSUE THREAD

**Issue by izuku-franck1555** [tada/pljava#197]
Hello everyone. I would like to install PL / JAVA but I am stuck at the last stage of the build which consists in executing "mvn clean install".
Here is a catch of the message that is presented to me and I am unfortunately a beginner. Thank you in advance for your help.
![capture](https://user-images.githubusercontent.com/43177617/46921141-61d0b000-cfef-11e8-9f56-e72c6c74eeee.PNG)

**Comment by jcflack** [comment]
Hello,

I've just done a complete rebuild starting from an empty Maven repository and using Maven 3.5.4 (the version you seem to be running) and it worked, so I think something in your Maven settings may be off. Have you used Maven in the past to build other projects? Do you have a `.m2` folder (possibly hidden) in your home folder, and does it contain a `repository` folder and/or a `settings.xml` file?

If so, I would try removing that `.m2` folder (or just renaming it so Maven does not find it) and try the build again. Your messages suggested that it had a problem downloading the `nar-maven-plugin` but it had no such problem for me just now.

A couple more notes: in the screenshot you provided, it looks like you really typed `mvn clean instal` with only one `l`. I do not think that was the cause of your issue, but it might have led to an unrecognized-goal error later.

I recommend that you do `git checkout V1_5_1b3` so that you are building the most recent, 1.5.1-BETA3, release. You are currently building from the tip of the 1.5 branch, which is very similar to the release but not exactly.

Regards,
-Chap

**Comment by izuku-franck1555** [comment]
Thank you for this hyper fast response @jcflack !
It's my first time using maven apache. So if I understand correctly the best is to resume everything to zero but removing m2 if there is, and change git checkout?
By the way, I want to say that I use a very old version of postgresql. This is 8.3. I tried to install PL / JAVA 1.4 but it was too complex. I did not understand anything in the documentation. So if you have some time I would like you to explain how to install it?

**Comment by jcflack** [comment]
Oh, you will find installing PL/Java 1.5.1 to be much simpler than 1.4 into your PostgreSQL 8.3 database. So I would continue trying to get 1.5.1 (beta3) built. Once you have it built, you can try installing it and report if you have any trouble.

If I may ask, what requires you to use such an old PG version?

**Comment by izuku-franck1555** [comment]
OK it is understood @jcflack . I will use 1.5.1. I am making the changes you have asked me to make. I send you the catches in a few minutes.
To answer your question, I wanted to use 1.4 because I thought it went with postgresql 8.3. Both were present in the same file. Thank you again for you help. It's been two days since I'm doing research on "how to install pl / java"

**Comment by izuku-franck1555** [comment]
I am using postgresql 8.3 as part of a class with an old teacher. He recommended us to install it because it is the one he uses. But when I have enough ideas I think I'll use a newer version.

**Comment by izuku-franck1555** [comment]
I always have the same mistakes.
![capture1](https://user-images.githubusercontent.com/43177617/46922183-41a7ed80-cffd-11e8-9ffb-413f43adf2a7.PNG)
![capture2](https://user-images.githubusercontent.com/43177617/46922184-42408400-cffd-11e8-99e0-9841f753562c.PNG)

**Comment by jcflack** [comment]
I think that there is something, like a previous partial or failed download, stuck in your local Maven repository. Did you find the `repository` folder and `settings.xml` and remove or rename them?

Ordinarily, the _first_ time you run `mvn` to build the package, you will see _hundreds_ of lines like

    Downloading from central: https://repo.maven.apache.org/maven2/org/...

the first build actually takes longer by several minutes because of all the downloading. After that, later builds go much faster, because the dependencies are already downloaded.

I do not see those hundreds of download lines, so I think you have a Maven repository already, and something funny is in it.

**Comment by izuku-franck1555** [comment]
You are right indeed. These lines are never posted at home. Here is how I proceed:
1- I download PL / JAVA via "git clone"
2- I position myself on it and execute "git checkout"
3- Then I run "mvn install"
Here is a screen of my local Apache Maven directory. There are no m2 or settings.xml. Or it's not my local repertoire, no idea. I was just happy to download apache maven and add it to my PATH.

**Comment by izuku-franck1555** [comment]
The screen of PL/JAVA folder and Apache maven folder
![capture3](https://user-images.githubusercontent.com/43177617/46922618-7ae35c00-d003-11e8-999b-fb8cc7df53b1.PNG)

**Comment by jcflack** [comment]
The local `repository` and `settings.xml` are typically in a folder called `.m2` that may be under `\Users\<your username>` or perhaps `\Documents and Settings\<your username>`.

You might easily find where they are with this command:

    mvn --debug --file nopom

It will end in an error (not finding a file `nopom`) but along the way there should be many debug messages. You can look for these:

    [DEBUG] Reading user settings from ...
    ...
    [DEBUG] Using local repository at ...

Once you know where that repository folder and settings file are, try making them go away.

Or, if they are not there, or empty, you can look for other `DEBUG` lines about the _global_ settings. If this is for a class, is it possible the instructor has made some settings already?

**Comment by izuku-franck1555** [comment]
I hope you can continue to help me tomorrow. Have a nice evening @jcflack .

**Comment by izuku-franck1555** [comment]
Hello @jcflack .
I finally found the folder "m2". It just contains the folder "repository", it does not have "settings.xml", I deleted it but I still have the same problem. I think the problem comes from the fact that I may have missed a step with Maven Apache. But because of my lack of experience in this area I really do not know where there can be error. Do you have another idea please?
![capture](https://user-images.githubusercontent.com/43177617/46978552-2a353700-d0c7-11e8-853d-652a2525ab9d.PNG)

**Comment by izuku-franck1555** [comment]
Another thing that confirms my idea is the fact that the "mvn --debug --file nopom" command did not display the lines that were intended as you can see in the capture.
![capture2](https://user-images.githubusercontent.com/43177617/46978803-c95a2e80-d0c7-11e8-8ed7-3017498c3819.PNG)

**Comment by jcflack** [comment]
On 10/15/2018 05:15 PM, izuku-franck1555 wrote:
> Another thing that confirms my idea is the fact that the "mvn --debug --file nopom" 

My apologies. I added "--file nopom" as a way to make Maven fail just
shortly after printing the needed lines. It works in older Maven
versions but I didn't test it in 3.5. They added an earlier test for
the pom file there so now it stops before giving any useful information.

You may as well try the plain command

  mvn --debug

while you are in some folder that does not contain any *.pom file.
That should limit the output to 100 lines or so. The "Reading user
settings from" and/or "Using local repository at" lines will be
shortly before the "BUILD FAILURE".

. - . - . - . - .

In any case, your earlier screenshot today at last shows what is
happening:

  Could not transfer artifact
com.github.maven-nar:nar-maven-plugin:pom:3.2.3 from/to central
(https://repo.maven.apache.org/maven2): repo.maven.apache.org: Unknown
host repo.maven.apache.org -> [Help 2]

So from your computer, Maven was not able to reach Maven Central on
the network to download a dependency. You may have to check what is
interfering with that network connection. If you run a web browser
on that computer, can you connect to the URL
  https://repo.maven.apache.org/maven2   ?

What does the command

  nslookup repo.maven.apache.org

show?

If this is for a class, is there someone on the technology staff
there who can help with troubleshooting the network connection?

-Chap

**Comment by izuku-franck1555** [comment]
Hello @jcflack !!
You were right, my problem is not "mvn clean install" but it's Apache maven. After reading your last answer I tried to run a basic command. This is "mvn archetype: generate" and it too failed. So it must be my connection, yet I easily downloaded PL / JAVA via the cmd, I also open the site https://repo.maven.apache.org/maven2. So I really do not see what level is the problem with maven?
[captur](https://user-images.githubusercontent.com/43177617/47039193-45fa1500-d17b-11e8-8af0-07db2bcfc410.PNG)
!

**Comment by izuku-franck1555** [comment]
Hello @jcflack !
You were right, my problem is not "mvn clean install" but it's Apache maven. After reading your last answer I tried to run a basic command. This is "mvn archetype: generate" and it too failed. So it must be my connection, yet I easily downloaded PL / JAVA via the cmd, I also open the site https://repo.maven.apache.org/maven2. So I really do not see what level is the problem with maven?
![captur](https://user-images.githubusercontent.com/43177617/47039307-907b9180-d17b-11e8-94d7-03fffd642c1d.PNG)

**Comment by izuku-franck1555** [comment]
Ah I forgot, I use jdk 1.7. Can that be the cause of the problem?

**Comment by jcflack** [comment]
On 10/16/2018 02:43 PM, izuku-franck1555 wrote:
> Hello @jcflack !
> You were right, my problem is not "mvn clean install" but it's Apache maven. After reading your last answer I tried to run a basic command. This is "mvn archetype: generate" and it too failed. So it must be my connection, yet I easily downloaded PL / JAVA via the cmd, I also open the site https://repo.maven.apache.org/maven2. So I really do not see what level is the problem with maven?

Argh, I think I could have seen the problem in your first comment
if I had looked more closely.

    Java version: 1.7.0_80, ...
    ...
    [ERROR] ... Could not transfer artifact
com.github.maven-nar:nar-maven-plugin:pom:3.2.3 from/to central
(https://repo.maven.apache.org/maven2): Received fatal alert:
protocol_version


As time goes on, some of the older encryption protocols used for secure
internet connections get replaced with newer ones, and then some web
sites will stop allowing the older protocols. It seems that
Maven Central will only allow connections using some of the more
modern protocol versions, and now it does not allow any of
the versions that were built into Java 7.

So it is not really a Maven problem, only that Maven is trying to use
the Java networking API to connect to Maven Central, and cannot, because
the Java 7 runtime does not support any protocol version new enough.

You can connect with your web browser, because it is newer than Java 7
and is able to use the newer protocols.

If you download a Java 8 JDK (either from

https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
or https://adoptopenjdk.net/releases.html#x64_win )

and use that to run Maven, it should work.

This is a bit frustrating, because I work to make sure PL/Java can
still be built with Java 8, 7, or 6. It works if you already have
all of the dependencies downloaded. But if Maven needs to connect
to the central repository, now only Java 8 is new enough....

Sorry I did not see the cause earlier.

-Chap

**Comment by izuku-franck1555** [comment]
Ok @jcflack.
I download and install jdk right away.

**Comment by izuku-franck1555** [comment]
It is working !!!
Thank you very much for your time. Thank you @jcflack !
![captu](https://user-images.githubusercontent.com/43177617/47043290-2d8ef800-d185-11e8-9870-1a7ba05b422e.PNG)

**Comment by izuku-franck1555** [comment]
Another mistake !!!
![capture](https://user-images.githubusercontent.com/43177617/47043717-37fdc180-d186-11e8-929d-5c977296e187.PNG)

**Comment by izuku-franck1555** [comment]
I went through the old mistake, but this one looks more serious.
![capture](https://user-images.githubusercontent.com/43177617/47046170-78603e00-d18c-11e8-88dd-499f2ee21c7e.PNG)

**Comment by jcflack** [comment]
On 10/16/2018 04:43 PM, izuku-franck1555 wrote:
> nar-validate ... Cannot deduce version number from: ...

Sounds like:

https://github.com/tada/pljava/issues/32#issuecomment-141865511

Do you have Microsoft Visual C++ installed on that machine?

Are you following the steps in:

http://tada.github.io/pljava/build/buildmsvc.html ?

Perhaps I should ask a question one step further back: do you know
whether the PostgreSQL instance you are building for was built
using MSVC, or using MinGW ?

If you go into psql and type SELECT version();
what does it say?

If it says MinGW instead of MS, then the appropriate instructions
would be these instead: http://tada.github.io/pljava/build/mingw64.html

**Comment by izuku-franck1555** [comment]
It is Visual C ++. Well, it's already installed.
![capture](https://user-images.githubusercontent.com/43177617/47048092-4b625a00-d191-11e8-8a24-31d0b8959fec.PNG)

**Comment by jcflack** [comment]
On 10/16/2018 05:17 PM, izuku-franck1555 wrote:
> It is Visual C ++. Well, it's already installed.

Ok, these should be the right instructions in that case:
http://tada.github.io/pljava/build/buildmsvc.html

The command window where you are running Maven, is it one that
you opened using a link from `Visual Studio Tools` from the
Start menu?

**Comment by izuku-franck1555** [comment]
No no I have never used a visual studio tool since I installed it.

**Comment by jcflack** [comment]
On 10/16/2018 05:33 PM, izuku-franck1555 wrote:
> No no I have never used a visual studio tool since I installed it.

Somewhere on the Start menu there should be some Visual Studio
tools (forgive my vagueness, I am not a Windows user myself) and
one of them should be to open a command window. If you are running
Maven in an ordinary command window (not started through Visual
Studio), Maven will not know how to find the compiler.

I notice the instructions say to match the MSVC version that
was used to build PostgreSQL. I do not even know how to find
out what MSVC version that was for PG 8.3. Perhaps pg_config
will tell you. (I wonder if you can even get that MSVC version!)

Possibly, it would save you a good deal of trouble to avoid
the PG 8.3 and simply download a more recent Windows PostgreSQL
build, perhaps from

https://www.openscg.com/bigsql/postgresql/installers/

I believe they already have PL/Java built, at least the 1.5.0
version, if you download their PostgreSQL 9.5 build.

https://www.openscg.com/bigsql/docs/proclang/proclang/#pljava

The PL/Java 1.5.1 beta that you are trying to build now will
very shortly become 1.5.1 official (as in, today or tomorrow),
and I can alert the BigSQL team to offer that as a build also,
so perhaps it will not be many days before it is available from
them. You would then be able to use their PostgreSQL 9.6 or 10
versions.

**Comment by izuku-franck1555** [comment]
Ok at first I will install PostgreSQL 9.5. I will explain to my teacher that I could not do otherwise. He will surely understand. Visual Studio did not recognize version 8.3.
![capture2](https://user-images.githubusercontent.com/43177617/47050424-3e953480-d198-11e8-8194-8acb032fc6c7.PNG)

**Comment by izuku-franck1555** [comment]
But finally I'm a little lost there, I have to install version 9.5 or 9.6?

**Comment by izuku-franck1555** [comment]
I opt for version 9.5, What changes will I have to make?
I will probably have to change the "git checkout", which therefore must include the suppresion file m2. That's right ?

**Comment by jcflack** [comment]
On 10/16/18 18:38, izuku-franck1555 wrote:
> I opt for version 9.5, What changes will I have to make?

Oh, if you download PostgreSQL 9.5 from BigSQL, you can just install
their PL/Java 1.5.0 also. They have already built it. Boom! You're in
business. :)

... unless, actually building PL/Java yourself is a requirement in
your class ...

**Comment by izuku-franck1555** [comment]
Ok I see. But the problem is that our teacher also wanted to test our skills from this installation. So using a pre-installed version will make me a little under. I think the best is to install PL / JAVA 9.1 and the Visual Studio that goes with it (the 10.0) and use a compatible PL / JAVA version. Since it's an old school class, I do not think it's necessary to install a recent version of PL / JAVA. For my research cons I have the setup of Postgresql 11 that I will be happy to install once my course finished. What do you think?
In summary I want to install Postgresql 9.1, Visual Studio 2010 and a corresponding PL / JAVA.

**Comment by izuku-franck1555** [comment]
I finally succeeded. Thank you for everything @jcflack  . Without you I will not have advanced.
I hope you can help me in the future and good luck for your research.
![capture](https://user-images.githubusercontent.com/43177617/47113803-db67d880-d251-11e8-968d-d13f303c09b8.PNG)

**Comment by jcflack** [comment]
On 10/17/2018 04:16 PM, izuku-franck1555 wrote:
> I finally succeeded.

Excellent! One thing, though: I see in your screenshot that you
have built some 1.5.1-SNAPSHOT source.

The official release 1.5.1 is now available. I strongly encourage
you to use it, and not to work with snapshot builds.

You can use the exact 1.5.1 sources here:

https://github.com/tada/pljava/releases/tag/V1_5_1

Or, if you have a git clone, you can do a git pull and then exactly

  git checkout V1_5_1

and rebuild. (Now that Maven has completed the necessary downloads
already, you will find rebuilding PL/Java takes very little time.)
You will see that the word SNAPSHOT no longer appears in your build.

Using the released version means you will be able to do
ALTER EXTENSION UPDATE; later when new PL/Java versions are
released. That command will not work if you have installed a
snapshot, because the source snapshots are continually changing
and do not identify a specific known version.

Regards,
-Chap

**Comment by izuku-franck1555** [comment]
Ok I changed version as you advised me. But I am facing a new difficulty and I do not know how to solve it. Here it is in this screenshot.
![capture](https://user-images.githubusercontent.com/43177617/47121268-f0e8fc80-d269-11e8-92c9-1b1d6ac41065.jpg)
![capture1](https://user-images.githubusercontent.com/43177617/47121269-f21a2980-d269-11e8-84db-a7632774d485.PNG)
![capture2](https://user-images.githubusercontent.com/43177617/47121270-f2b2c000-d269-11e8-9127-7522a208f157.PNG)

**Comment by jcflack** [comment]
On 10/17/2018 07:08 PM, izuku-franck1555 wrote:
> Ok I changed version as you advised me. But I am facing a new difficulty 

That is to be expected; PL/Java needs to be told what Java runtime
it should use. That is done with the PostgreSQL configuration variable
pljava.libjvm_location, which you should set to the full path of the
jvm.dll file in your installed Java runtime.

The documentation covering that is on these pages:

http://tada.github.io/pljava/install/install.html
http://tada.github.io/pljava/install/locatejvm.html

For example, if you find the file at
C:\Program Files\Java\jdk1.8.0_192\jre\bin\server\jvm.dll

then just use the psql command

SET pljava.libjvm_location TO 'C:\Program
Files\Java\jdk1.8.0_192\jre\bin\server\jvm.dll';

If your psql session is still at the prompt where it ends in your
screenshot, after you SET the variable, it should continue installing
immediately. The result would be that PL/Java is installed, but in
the old (before PostgreSQL had "extensions") way. It will show a
message instructing you what to do about that. :)

Or, if you have already exited that psql session, just start a new
one, do the SET pljava.libjvm_location TO ... command first, then
CREATE EXTENSION pljava; and it should Just Work. :)

It is important that, once you have the setting for
pljava.libjvm_location that works, you save that setting in
PostgreSQL. Otherwise, the next time you start a new session,
PL/Java won't work.

You can use ALTER DATABASE postgres SET pljava.libjvm_location FROM
CURRENT; and it will save the current (working) value, so that later
sessions have it.

One more note: while you used Java 8 to build PL/Java, you are not
forced to the same version when using it. You could download
Java 11 and install it, and set pljava.libjvm_location to that
jvm.dll. That way, when you are writing your own database code
in PL/Java, you can use the latest Java 11 language features.

-Chap

**Comment by izuku-franck1555** [comment]
It's good now. I do not really know how I did it but it seems that everything is working properly. Luck :)
![capture-final](https://user-images.githubusercontent.com/43177617/47146899-ed369380-d2c4-11e8-9164-93f9082469aa.PNG)

**Comment by jcflack** [comment]
Hi,

This issue was closed nearly six years ago, involved a previous major version of PL/Java, and a Maven plugin that the current PL/Java version's build system doesn't use. So it seems unlikely that your issue is really the same.

Could you open a new issue, and include your OS and version, PostgreSQL, Java, and PL/Java versions, and the output that indicates a problem?

Thanks!

**Comment by saee-tech** [comment]
Scanning for projects...
[INFO] 
[INFO] --------------------------< com.example:demo >--------------------------
[INFO] Building DevOps 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.3.2:clean (default-clean) @ demo ---
[INFO] 
[INFO] --- maven-toolchains-plugin:3.1.0:toolchain (default) @ demo ---
[INFO] Required toolchain: jdk [ vendor='openjdk' version='21' ]
[INFO] Found matching toolchain for type jdk: JDK[/usr/lib/jvm/java-21-openjdk-amd64]
[INFO] 
[INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ demo ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.966 s
[INFO] Finished at: 2025-01-20T13:39:58Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-resources-plugin:3.3.1:resources (default-resources) on project demo: Cannot create resource output directory: /home/ubuntu/ResumeApplication/target/classes -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException





I am getting above error while using "mvn clean Install"
